usercp.php

```diff
@@ -3221,15 +3221,18 @@
 
 			$db->insert_query("joinrequests", $joinrequest);
 
-			foreach($groupleaders[$usergroup['gid']] as $leader)
+			if(in_array($usergroup['gid'], $groupleaders))
```

Comment:
The array has this structure: `$groupleaders[$gid][$uid] = $user;` So wouldn't `in_array($gid, $groupleaders)` also check the `$uid` part? Probably better (and definitly more intuitive) would be `is_array($groupleaders[$gid])`.